gene_gene/gene_interacts_gene/dataset_statistics/dataset_statistics.py

```diff
@@ -0,0 +1,520 @@
+    inner join candidate on gene_gene.id=candidate.id 
+    where split={}
+) as candidate_splits inner join span on candidate_splits."Gene1_id"=span.id;
+'''
+
+
+# In[9]:
+
+
+train_candidate_df = pd.read_sql(sentence_sql.format(3), database_str)
+train_candidate_df.head(2)
+
+
+# In[10]:
+
+
+dev_candidate_df = pd.read_sql(sentence_sql.format(4), database_str)
+test_candidate_df = pd.read_sql(sentence_sql.format(5), database_str)
+
+
+# In[11]:
+
+
+clean_up_df = lambda x: (
+        entity_stats_df
+        .merge(sentence_count_df, on="sentence_id")
+        .merge(x, on="sentence_id")
+        .rename(index=str, columns={
+            "disease":"disease_mention_count", 
+            "gene":"gene_mention_count", 
+            "compound":"compound_mention_count",
+            "cand_id": "candidate_id"
+        })
+    )
+
+
+# In[12]:
+
+
+train_candidate_df = clean_up_df(train_candidate_df)
+train_candidate_df.head(2)
+
+
+# In[13]:
+
+
+dev_candidate_df = clean_up_df(dev_candidate_df)
+test_candidate_df = clean_up_df(test_candidate_df)
+
+
+# In[14]:
+
+
+training_set_df = (
+    entity_level_df
+    .query("split==3&has_sentence==1")
+    .merge(
+        train_candidate_df.astype({"gene1_id":int, "gene2_id":int}), 
+        on=["gene1_id", "gene2_id"]
+    )
+)
+training_set_df.head(2)
+
+
+# In[15]:
+
+
+dev_set_df = (
+    entity_level_df
+    .query("split==4&has_sentence==1")
+    .merge(
+        dev_candidate_df.astype({"gene1_id":int, "gene2_id":int}), 
+        on=["gene1_id", "gene2_id"]
+    )
+)
+
+test_set_df = (
+    entity_level_df
+    .query("split==5&has_sentence==1")
+    .merge(
+        test_candidate_df.astype({"gene1_id":int, "gene2_id":int}), 
+        on=["gene1_id", "gene2_id"]
+    )
+)
+
+
+# In[16]:
+
+
+total_candidates_df = (
+    training_set_df
+    .append(dev_set_df)
+    .append(test_set_df)
+    .drop_duplicates("candidate_id")
+)
+
+
+# In[ ]:
+
+
+dev_candidates = (
+    session
+    .query(GeneGene)
+    .filter(
+        GeneGene.id.in_(
+            total_candidates_df
+            .query("split==4")
+            .sample(10000, random_state=100)
+            .candidate_id
+            .tolist()
+        )
+    )
+    .all()
+)
+dev_df = make_sentence_df(dev_candidates)
+dev_df.head(2)
+
+
+# In[ ]:
+
+
+test_candidates = (
+    session
+    .query(GeneGene)
+    .filter(
+        GeneGene.id.in_(
+            total_candidates_df
+            .query("split==5")
+            .sample(10000, random_state=120)
+            .candidate_id
+            .tolist()
+        )
+    )
+    .all()
+)
+test_df = make_sentence_df(test_candidates)
+test_df.head(2)
+
+
+# In[ ]:
+
+
+write_candidates_to_excel(dev_df, "data/sentence_labels_dev.xlsx")
+write_candidates_to_excel(test_df, "data/sentence_labels_test.xlsx")
+
+
+# ## Distribution of Sentence Length
+
+# In[17]:
+
+
+sns.distplot(total_candidates_df["sen_length"], rug=False)
```

Comment:
This distribution looks a bit suspicious..do you know what its oscillating so much?  The regularity seems odd

Reply:
That is a good question. I'm not exactly sure why this oscillation is occurring. I think this distribution highlights a trend on how scientific authors write sentences containing gene-gene information.